ChangeLog.d/pkparse_public_key_memleak.txt

```diff
@@ -0,0 +1,3 @@
+Bugfix
```

Comment:
Please add a non-regression test: at least one case where `mbedtls_pem_read_buffer` succeeds and `mbedtls_pk_setup` fails. Our CI runs the unit tests under a memory leak detector, so there's no need for specific code to detect the memory leak.

Reply:
Are you referring to a test similar to the code snippet I have provided in issue #5492? That would be easy to implement, but would translate to having a test case that relies on an invalid use flow. If you are referring to a test that fails that particular memory allocation in `mbedtks_pk_setup`, that would be significantly harder to implement, but would capture the essence of this bug.

Reply:
Forcing a failure may require adding a test hook. I'll take care of making a test.